### ISSUE TITLE
Missing newly inserted rows in incrementing column mode (MySQL)

### ISSUE THREAD

**Issue by xjrk58** [confluentinc/kafka-connect-jdbc#915]
PROBLEM:

When reading from read-only MySQL replica it might happen that the duration of two transactions is different. When reading rows from append-only table with auto-increment primary key it might happen that we miss some inserts if we issue a query at the wrong moment. 

Example:
Transaction A 
Begin logical time: 1
Commit logical time: 10

Transaction B
Begin logical time: 2
Commit logical time: 5

Select query by Kafka Connect JDBC connector at logical time 6. Because transaction B auto-increment generates higher id than for transaction A, transaction A will be ignored and never read by the connector.

**Comment by xjrk58** [comment]
@ncliang I hope the last edit brings a bit more clarity into why this is an issue.

**Comment by ncliang** [comment]
Ah, thanks for the context @xjrk58 . Could you set the `poll.interval.ms` higher to workaround the issue? I think the PR that you proposed special cases this and waits for an additional `poll.interval.ms` . I am worried that even with that change, the core of the issue is still present, a transaction that is in flight for more than 2x `poll.interval.ms` would still get missed. This is just due to the polling nature of the connector. Unless you take a CDC approach like Debezium, the issue would still remain.

**Comment by xjrk58** [comment]
Setting `poll.interval.ms` helps, but it doesn't eliminate the issue. It only decreases probability of missing the data. We tested approach proposed in the linked PR and the probability of missing some rows goes effectively to zero for non-batch inserts with Aurora/MySQL. If `poll.interval.ms` is set to maximum transaction time + maximum processing time then no transactions should be missed. The only downside of this approach is that some constant delay is introduced.

**Comment by ncliang** [comment]
@xjrk58 , yes, the workaround would introduce additional delay due to the increased poll interval. But, I also don't think it is worth special casing to handle in code since the special case handling is also just a slightly better workaround that doesn't resolve the root cause. If you are running into this edge case frequently and don't want to workaround with increasing delay interval, the proposed PR workaround likely also wouldn't work. And, the only true solution is to not use the poll-based JDBC connector and switch to a CDC-based connector.

**Comment by xjrk58** [comment]
@ncliang actually this is not a special case. What typically happens is that the time between beginning of the transaction and commit in the source database is not uniform, so even if you delay your reads you would see a strong variations in the timestamp and simply by miss some changes with near certainty.

With approach the following approach it's possible to smoothen those variants and capture nearly all changes.  
1. read max timestamp, store in memory (per task)
2. wait until the next poll
3. use stored max timestamp in the query instead of current timestamp
4. repeat
